oak_loader/src/options.rs

```diff
@@ -136,13 +143,17 @@ pub fn create_runtime_config() -> anyhow::Result<oak_runtime::RuntimeConfigurati
         } else {
             None
         },
-        introspect_port: if cfg!(feature = "oak_debug") && !opt.no_introspect {
+        introspect_port: if cfg!(feature = "oak_debug")
+            && !opt.no_introspect
+            && permissions_config.allow_introspection
```

Comment:
I think its a non-obvious change that is also hidden inside the code.
Should we have two separate permissions files, and choose between them in `runner` based on `oak_debug` ?

Reply:
`oak_debug` is not a flag, it is a compile-time feature, it is "provided" by whoever compiles the code. The idea is that enabling `oak_debug` results in a version of Oak that does *not* enforce IFC (maybe we should rename the feature to `unsafe_oak_debug` or something scarier), while defense-in-depth permissions should result in a safe execution always. In particular, if introspection leaks data bypassing IFC, then it should probably only be available in `oak_debug` (and at that point perhaps there is not much point in also having it in the permission file).

cc @daviddrysdale

Reply:
Thanks for the clarification @tiziano88. 
If there is no flag for disabling introspection in the permissions file, then defense-in-depth permissions will not result in a safe execution if the code is compiled with `oak_debug`. Let's say someone, by mistake, deploys a version that is compiled with `oak_debug`. With a flag for introspection in the permissions file, there is still a way to disable introspection. So, I think it makes sense to have it in the permissions file. The same applies to the log nodes too.

Reply:
If someone deploys a debug version of Oak to production, we have much bigger problems :) since that could literally just log everything out.

Reply:
I'm not against this change, but the way it is currently structured seems confusing, since multiple flags (some compile time, some run time) control the same underlying functionality in slightly different ways (e.g. logging, introspection). I think we should have a clearer explanation of how these things interact.

Reply:
Oh! Good! I have always been wondering about the logs!! Now, I see that we actually disable logging when not compiled with `oak_debug`, and the log node sends to stderr. If the permissions-file is meant to be used with non-debug versions, then I agree that there is no point for having a permission for introspection.

Reply:
Pushed a new change that removes `allow_introspection`, and limits the use of the permissions file to logless servers (those compiled with `oak_debug` disabled). WDYT?